pep-0xxx.rst

```diff
@@ -0,0 +1,1250 @@
+Instead, a small sample of entries is provided to give a flavor of how it will
+appear.
+
+::
+
+    class CipherSuite(Enum):
+        ...
+        TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA = (0xC0, 0x12)
+        ...
+        TLS_ECDHE_ECDSA_WITH_AES_128_CCM = (0xC0, 0xAC)
+        ...
+        TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 = (0xC0, 0x2B)
+        ...
+
+
+Protocol Negotiation
+~~~~~~~~~~~~~~~~~~~~
+
+Both NPN and ALPN allow for protocol negotiation as part of the HTTP/2
+handshake. While NPN and ALPN are, at their fundamental level, built on top of
+bytestrings, string-based APIs are frequently problematic as they allow for
+errors in typing that can be hard to detect.
+
+For this reason, this module would define a type that protocol negotiation
+implementations can pass and be passed. This type would wrap a bytestring to
+allow for aliases for well-known protocols. This allows us to avoid the
+problems inherent in typos for well-known protocols, while allowing the full
+extensibility of the protocol negotiation layer if needed by letting users pass
+byte strings directly.
+
+::
+
+    class NextProtocol(Enum):
+        H2 = b'h2'
+        H2C = b'h2c'
+        HTTP1 = b'http/1.1'
+        WEBRTC = b'webrtc'
+        C_WEBRTC = b'c-webrtc'
+        FTP = b'ftp'
+        STUN = b'stun.nat-discovery'
+        TURN = b'stun.turn'
+
+TLS Versions
+~~~~~~~~~~~~
+
+It is often useful to be able to restrict the versions of TLS you're willing to
+support. There are many security advantages in refusing to use old versions of
+TLS, and some misbehaving servers will mishandle TLS clients advertising
+support for newer versions.
+
+The following enumerated type can be used to gate TLS versions. Forward-looking
+applications should almost never set a maximum TLS version unless they
+absolutely must, as a TLS backend that is newer than the Python that uses it
+may support TLS versions that are not in this enumerated type.
+
+Additionally, this enumerated type defines two additional flags that can always
+be used to request either the lowest or highest TLS version supported by an
+implementation.
+
+::
+
+    class TLSVersion(Enum):
+        MINIMUM_SUPPORTED
+        SSLv2
+        SSLv3
+        TLSv1
+        TLSv1_1
+        TLSv1_2
+        TLSv1_3
+        MAXIMUM_SUPPORTED
+
+
+Errors
+~~~~~~
+
+This module would define three base classes for use with error handling. Unlike
+many of the the other classes defined here, these classes are not abstract, as
+they have no behaviour. They exist simply to signal certain common behaviours.
+Backends should subclass these exceptions in their own packages, but needn't
+define any behaviour for them.
+
+In general, concrete implementations should subclass these exceptions rather
+than throw them directly. This makes it moderately easier to determine which
+concrete TLS implementation is in use during debugging of unexpected errors.
+However, this is not mandatory.
+
+The definitions of the errors are below::
```

Comment:
I think you want a way to distinguish TLS configuration errors (perhaps inheriting from ValueError?) from actual I/O and/or crypto-related errors (which should inherit from OSError, as SSLError currently does).